### PR TITLE
Woo-Connect Insights: use existing site-stats-module, refactor widget-list and styling

### LINKED ISSUE #15762: Woo-connect Insights: statsOrders Normalizer Refactor
statsOrders data response from normalizers needs to be refactored into array, splitting up data and deltas

### PR DESCRIPTION
**Pull request by greenafrican**
- removed redundant/duplicate code from `store-stats-widget-list` to use `store-stats-module`
- updated styling
- consolidated helper methods and added tests
- didn't quite get to refactor `statsOrders`, which is returned as an `Object` with data and deltas; ideally, there are 2 calls (1 for each) but these are coupled with the statType used to get the data from `getSiteStatsForQuery`

Fixes #15762

### CODE REVIEW

File: client/extensions/woocommerce/app/store-stats/index.js [hunk 1]
```diff
@@ -123,8 +115,18 @@ class StoreStats extends Component {
 					/>
 				</StatsPeriodNavigation>
 				<div className="store-stats__widgets">
-					{ widgetList1 }
-					{ widgetList2 }
+					<div className="store-stats__widgets-column spark-widgets" key="sparkwidgets">
+						<Module
+							siteId={ siteId }
+							header={ null }
+							emptyMessage={ translate( 'No data found.' ) }
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/store-stats/index.js]: Hi! I've found a possible matching string that has already been translated **22** times:
`translate( 'No data found' )` *ES Score: 10*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/fix/wcss-refactor-statsorders)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/app/store-stats/store-stats-widget-list/index.js [hunk 2]
```diff
@@ -74,111 +55,95 @@ class StoreStatsWidgetList extends Component {
...
+				key: 'title',
+				label: translate( 'Stat' )
+			},
+			{
+				key: 'value',
+				label: translate( 'Value' )
+			},
+			{
+				key: 'sparkline',
+				label: translate( 'Trend' )
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/store-stats/store-stats-widget-list/index.js]: Hi! I've found a possible matching string that has already been translated **34** times:
`translate( 'Trending' )` *ES Score: 13*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/fix/wcss-refactor-statsorders)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/app/store-stats/store-stats-widget-list/index.js [hunk 3]
```diff
@@ -74,111 +55,95 @@ class StoreStatsWidgetList extends Component {
...
-		let widgetList = null;
+		const { data, deltas, query, widgets } = this.props;
+		const selectedIndex = this.getSelectedIndex( data );
+		const firstRealKey = Object.keys( deltas[ selectedIndex ] ).filter( key => key !== 'period' )[ 0 ];
+		const sincePeriod = this.getDeltaByStat( firstRealKey );
+		const periodFormat = getPeriodFormat( query.unit, sincePeriod.reference_period );
+		const values = [
+			{
+				key: 'title',
+				label: translate( 'Stat' )
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/store-stats/store-stats-widget-list/index.js]: Hi! I've found a possible matching string that has already been translated **78** times:
`translate( 'Stats' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by westi** [comment]
I'm seeing some strange UI issues in the deltas with this change applied locally.

For example comparing the deltas on: https://wpcalypso.wordpress.com/store/stats/orders/week/westiwoo.mystagingwebsite.com?startDate=2017-05-14

screen shot 2017-07-11 at 08 30 19

To them with the patch applied: http://calypso.localhost:3000/store/stats/orders/week/westiwoo.mystagingwebsite.com?startDate=2017-05-14

screen shot 2017-07-11 at 08 30 57

The graph has gone outside the bounding box for some reason.

**Comment by greenafrican** [comment]
> graph has gone outside the bounding box

@westi I'm tracking the sparkline issue in another issue here - #16135 

Going to move forward with this.

**Pull request merged**